packages/components/src/components/AutoScalingInput/AutoScalingInput.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useRef, useEffect, forwardRef, useState, ChangeEvent } from 'react';
+import styled from 'styled-components';
+
+const HiddenInputToMeasurePlaceholderScrollableWidth = styled.input`
+    visibility: hidden !important;
+    width: 0 !important;
+    height: 0 !important;
+    /* Never change any style here! It would affect the scrollWidth calculation of the placeholder. */
+`;
+
+type ApplyWidthParams = {
+    ref: React.MutableRefObject<HTMLInputElement | null>;
+    width: number;
+    minWidth: number;
+};
+
+const applyWidth = ({ width, minWidth, ref }: ApplyWidthParams) => {
+    if (ref?.current?.style) {
+        const borderSize = ref.current.offsetWidth - ref.current.clientWidth;
+
+        // First it needs to be set to minimum, this ensures that with of the input will be shrinking as text gets smaller
+        ref.current.style.width = `${minWidth + borderSize}px`; // See: https://stackoverflow.com/a/75227086
+        ref.current.style.width = `${width + borderSize}px`;
+    }
+};
+
+type ApplyOnTargetOverEventParams = {
+    onChange: React.InputHTMLAttributes<HTMLInputElement>['onChange'];
+    calculatedMin: number;
+};
+
+const createHandleOnChangeAndApplyNewWidth =
+    ({ onChange, calculatedMin }: ApplyOnTargetOverEventParams) =>
+    (event: ChangeEvent<HTMLInputElement>) => {
+        const { target } = event;
+        const borderSize = target.offsetWidth - target.clientWidth;
+
+        // See: `applyWidth` function for explanation.
+        // It cannot be used here as we work with `target` from event and not react Ref.
+        target.style.width = `${calculatedMin + borderSize}px`; // See: https://stackoverflow.com/a/75227086
+        target.style.width = `${target.scrollWidth + borderSize}px`;
+        onChange?.(event);
+    };
+
+type Props = React.DetailedHTMLProps<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    HTMLInputElement
+> & { minWidth: number };
+
+export const AutoScalingInput = forwardRef<HTMLInputElement, Props>(
```

Comment:
Btw should we allow account names like " "? 😂

<img width="532" alt="image" src="https://github.com/trezor/trezor-suite/assets/858321/a3516cb5-4eeb-4ee4-8c47-2059b49be0ce">

Reply:
> Btw should we allow account names like " "? 😂

:heavy_check_mark:

Reply:
> This focus border radius is weird:

This is not happening to me. I am afraid this may be some default styling for Mac inputs? Lets discuss it on slack and we can test it together.